### ISSUE TITLE
Performance related Question

### ISSUE THREAD

**Issue by tugan** [awslabs/dynamodb-titan-storage-backend#54]
Hi,

we have single vertex and single edge very simple model and use dynamodb backend. increased table read capacity. 

when we try yo get around 3000 record from server it takes 20-25 second. it that normal? should we do any additional configuration changes?

regards
Burak Tugan

**Comment by amcp** [comment]
The DynamoDB Storage Backend for Titan self-throttles to avoid getting throttled on the service side. Getting 3000 small (<1KiB) items in 20 seconds suggests that your read capacity is set to no more than 150 RPS, assuming that you use the single-item data model, no vertex partitioning and assuming that you read each record using random access (as opposed to a random scan). After you share details about your access pattern, data model and partitioning settings, and the throughput configuration of your graph, I could help you some more.

**Comment by amcp** [comment]
Also, you should definitely turn on [query.fast-property=true](http://s3.thinkaurelius.com/docs/titan/1.0.0/titan-config-ref.html#_query), to pull all the properties of a vertex when it gets loaded (instead of lazy-loading them whenever they are referenced)

**Comment by amcp** [comment]
If you have any more questions about how to improve the query and traversal performance of Titan on DynamoDB, please re-open this issue.

**Comment by bendavidwhite** [comment]
Hello

I'm seeing similar performance issues to Burak. We also have a simple single vertex, single edge model. There are about 400 vertices loaded, each with anywhere between 5 and 100 connected edges.

Queries traversing the network and returning just a list of vertices or edges return in under a second. However when appending either `.valueMap()` or `.values()` to retrieve property values the query time increases by a factor of 5-10 times. There are only two properties and they are both short sub 10 character strings.

I've tried adding `query.fast-property=true` to dynamodb.properties as well as bumping up the values of the likes of `storage.dynamodb.stores.edgestore.initial-capacity-read` by a factor of 100. Similarly I've increased the read capacity units for the jg_ tables in the DynamoDB console.

I feel like there's something fundamental I'm missing as when inspecting the item counts of the jg_ tables they are all zero! See below.  This is despite always using the `:remote`  command so that requests to add vertices/edges etc are sent to the Gremlin Server. 

Thanks in advance for any help!
Ben

<img width="1089" alt="screen shot 2017-10-21 at 20 03 21" src="https://user-images.githubusercontent.com/32987465/31855068-fc72309e-b69a-11e7-84f2-7770f0f06a27.png">
